src/streaming/text/TextTracks.js

```diff
@@ -547,7 +547,7 @@ function TextTracks(config) {
 
                                 // If disabled, add fragment cues as they'll be downloaded once.
                                 // If you miss them, they won't be downloaded again.
-                                if (track.mode !== Constants.TEXT_DISABLED || (appendFragmentedCuesWhenTrackModeDisabled && isFragmented)) {
+                                if (track.mode !== Constants.TEXT_DISABLED || (appendFragmentedCuesWhenTrackModeDisabled && (isFragmented || track.kind === 'captions'))) {
```

Comment:
I wonder if this is a deeper issue, shouldn't `isFragmented` be `true` here?

Reply:
Since the subtitle was not selected, the `currentTrackIdx` is -1.  The return value of  `this.getCurrentTrackInfo()?.isFragmented` is false.
```
    function getCurrentTrackInfo() {
        return textTrackQueue[currentTrackIdx];
    }
```

Reply:
CC track is not isFragmented. Only text tracks explicitly defined in the adaptation, the value can be true.